ui/src/services/api.js

```diff
@@ -59,21 +77,70 @@ export async function createNode(payload) {
   const body = {
     metadata: {
       name: payload.name,
+      labels: {
+        'metalk8s.scality.com/version': '2.0'
+      },
       annotations: {
         'metalk8s.scality.com/ssh-user': payload.ssh_user,
         'metalk8s.scality.com/ssh-port': payload.ssh_port,
         'metalk8s.scality.com/ssh-host': payload.hostName_ip,
         'metalk8s.scality.com/ssh-key-path': payload.ssh_key_path,
-        'metalk8s.scality.com/ssh-sudo': payload.sudo_required.toString(),
-        'metalk8s.scality.com/workload-plane': payload.workload_plane.toString(),
-        'metalk8s.scality.com/control-plane': payload.control_plane.toString()
+        'metalk8s.scality.com/ssh-sudo': payload.sudo_required.toString()
       }
+    },
+    spec: {
+      taints: []
     }
   };
 
+  if (payload.workload_plane && payload.control_plane) {
+    body.metadata.labels[ROLE_MASTER] = '';
+    body.metadata.labels[ROLE_NODE] = '';
+    body.metadata.labels[ROLE_ETCD] = '';
+  } else if (!payload.workload_plane && payload.control_plane) {
+    body.metadata.labels[ROLE_MASTER] = '';
+    body.metadata.labels[ROLE_ETCD] = '';
+
+    body.spec.taints = body.spec.taints.concat([
+      {
+        key: ROLE_MASTER,
+        effect: 'NoSchedule'
+      },
+      {
+        key: ROLE_ETCD,
+        effect: 'NoSchedule'
+      }
+    ]);
+  } else if (payload.workload_plane && !payload.control_plane) {
+    body.metadata.labels[ROLE_NODE] = '';
+  }
+
   try {
     return await coreV1.createNode(body);
   } catch (error) {
     return { error };
   }
 }
+
+export async function deployNode(url, token, node) {
+  try {
+    return await axios.post(
+      url,
+      {
+        client: 'runner',
+        fun: 'state.orch',
+        arg: ['metalk8s.orchestrate.deploy_salt_minion_on_new_node'],
```

Comment:
This won't be sufficient, but there's not much the UI can do right now I believe. This should be a one-shot runner doing both `salt-minion` deployment as well as actually deploying the node.